src/main/java/seedu/duke/command/Command.java

```diff
@@ -86,6 +89,69 @@ public void execute(RecipeList recipeList, UI ui) throws IOException {
                 ui.showAddingRecipeErrorMessage(e);
             }
             break;
+        case ADDTORECIPE:
+            try {
+                if (fullDescription.isEmpty()) {
+                    throw new IncompleteInputException("The description of " + type + " cannot be empty.\n");
+                }
+                if (!Parser.isValidAddToRecipe(fullDescription)) {
+                    ui.showInvalidAddToRecipeDescription();
+                    break;
+                }
+                String[] parsed = Parser.parseAddToRecipeDescription(fullDescription);
+                if (parsed.length != 3) {
+                    ui.showInvalidAddToRecipeDescription();
+                    break;
+                }
+                String elementType = parsed[0];
+                String id = parsed[1];
+                String description = parsed[2];
```

Comment:
@YatPang Perhaps these should be trimmed first before you use them to compare and such in the later parts of this method, likewise for the deleteFromRecipe method.

Reply:
Trimmed in parseAddToRecipeDescription/parseDeleteFromRecipeDescription. Mark as resolved.